docs/datatypes/tset.md

```diff
@@ -0,0 +1,251 @@
+---
+id: datatypes_tset
+title: "TSet"
+---
+
+A `TSet[A]` is transactional data structure built on top of `TMap[A]`.
+
+## Create a TSet
+
+Creating an empty `TSet`:
+
+```scala mdoc:silent
+import zio._
+import zio.stm._
+
+val emptyTSet: STM[Nothing, TSet[Int]] = TSet.empty[Int]
+```
+
+Or creating a `TSet` with specified values:
+
+```scala mdoc:silent
+import zio._
+import zio.stm._
+
+val specifiedValuesTSet: STM[Nothing, TSet[Int]] = TSet.make(1, 2, 3)
+```
+
+Or creating a `TSet` from existing `Iterable` structure:
```

Comment:
Rephrase as `Alternatively, you can create a `TSet` by providing a collection of values:`. Here's the right place to put the remark about handling duplicates (last one "wins").